### PR TITLE
feat: bump the k8s version

### PR DESCRIPTION
**Pull request by SandhyaRavi2403**
**What problem does this PR solve?**:
bump k8s version to 1.30.5

**Which issue(s) does this PR fix?**:
https://jira.nutanix.com/browse/NCN-102271

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: ansible/group_vars/all/defaults.yaml [hunk 1]
```diff
@@ -5,17 +5,17 @@ python_path: ""
 # If it's not there, the kubernetes_full_version will have "None" for a version number.
 #
 # IMPORTANT When you update kubernetes_version, also update crictl_version.
-kubernetes_version: "1.29.6"
+kubernetes_version: "1.30.5"
 kubernetes_major_minor: "{{ (kubernetes_version.split('.') | map('trim'))[:2] | join('.') }}"
 kubernetes_semver: "v{{ kubernetes_version }}"
 
-kubernetes_cni_version: "1.3.0"
+kubernetes_cni_version: "1.5.1"
```
Comment (dkoshkin) [inline ansible/group_vars/all/defaults.yaml:12]: I think it should be like this, I'm curious if you saw `1.5.1` somewhere else?
```suggestion
kubernetes_cni_version: "1.4.0"
```

File: ansible/group_vars/all/defaults.yaml [hunk 2]
```diff
@@ -5,17 +5,17 @@ python_path: ""
...
 
 # The crictl CLI is released as part of the http://sigs.k8s.io/cri-tools project.
 # The project release closely follows the Kubernetes release cycle, and uses a
 # nearly identical version scheme.
 # IMPORTANT When you update crictl_version, also update crictl_sha256.
-crictl_version: "{{ kubernetes_major_minor }}.0"
+crictl_version: "{{ kubernetes_major_minor }}.1"
```
Comment (dkoshkin) [inline ansible/group_vars/all/defaults.yaml:18]: Yes I think you're right here it should be .1
When I run the install from the upstream repo, I see
```
 cri-tools aarch64 1.30.1-150500.1.1
 ```
In that case we need to also update https://github.com/mesosphere/dkp-release/blob/175d9d167cd5015ed26f430b29463eef1ea65907/k8s-build/package_rpms.sh#L44-L53

File: ansible/group_vars/all/defaults.yaml (hunk 1 above)
Comment (SandhyaRavi2403) [reply ansible/group_vars/all/defaults.yaml:12]: I did refer the below link based on https://github.com/mesosphere/konvoy-image-builder/blob/73172e0a5abc96ac936da338b3b94b6dad3ffd97/ansible/group_vars/all/defaults.yaml#L69

https://github.com/containernetworking/plugins/releases

File: ansible/group_vars/all/defaults.yaml (hunk 2 above)
Comment (SandhyaRavi2403) [reply ansible/group_vars/all/defaults.yaml:18]: @dkoshkin PR: https://github.com/mesosphere/dkp-release/pull/145

File: ansible/group_vars/all/defaults.yaml (hunk 1 above)
Comment (SandhyaRavi2403) [reply ansible/group_vars/all/defaults.yaml:12]: @dkoshkin So do we have to change the cni version to 1.4.0 ?

File: ansible/group_vars/all/defaults.yaml (hunk 2 above)
Comment (dkoshkin) [reply ansible/group_vars/all/defaults.yaml:18]: We want to make another change in https://github.com/mesosphere/konvoy-image-builder/blob/73172e0a5abc96ac936da338b3b94b6dad3ffd97/Dockerfile.devkit#L90 

```
export CRICTL_TOOLS_VERSION="$(echo ${KUBERNETES_VERSION} | cut -d. -f1-2).1"
```

Comment (SandhyaRavi2403) [reply ansible/group_vars/all/defaults.yaml:18]: done

Comment (dkoshkin) [reply ansible/group_vars/all/defaults.yaml:18]: One more for the FIPS packages https://github.com/mesosphere/konvoy-image-builder/blob/c3c49c5467207b79f30b30c2d26d8b807bce65e4/Dockerfile.devkit#L103

Comment (dkoshkin) [reply ansible/group_vars/all/defaults.yaml:18]: For the Ubuntu test, 
It should be:
```
Get:1 https://prod-cdn.packages.k8s.io/repositories/isv:/kubernetes:/core:/stable:/v1.30/deb  cri-tools 1.30.1-1.1
```

Please update it here and correct the comment.
https://github.com/mesosphere/konvoy-image-builder/blob/73172e0a5abc96ac936da338b3b94b6dad3ffd97/cmd/konvoy-image-wrapper/cmd/create-package-bundle.go#L295-L296

File: ansible/group_vars/all/defaults.yaml [hunk 3]
```diff
@@ -5,17 +5,17 @@ python_path: ""
...
 
-kubernetes_cni_version: "1.3.0"
+kubernetes_cni_version: "1.4.0"
 
 # The crictl CLI is released as part of the http://sigs.k8s.io/cri-tools project.
 # The project release closely follows the Kubernetes release cycle, and uses a
 # nearly identical version scheme.
 # IMPORTANT When you update crictl_version, also update crictl_sha256.
-crictl_version: "{{ kubernetes_major_minor }}.0"
+crictl_version: "{{ kubernetes_major_minor }}.1"
```
Comment (dkoshkin) [inline ansible/group_vars/all/defaults.yaml:18]: On line 24 the SHA needs to change too 

https://github.com/mesosphere/konvoy-image-builder/actions/runs/11039503484/job/30666394971?pr=1188#step:9:1368

File: ansible/group_vars/all/defaults.yaml (hunk 2 above)
Comment (SandhyaRavi2403) [reply ansible/group_vars/all/defaults.yaml:18]: done

File: ansible/group_vars/all/defaults.yaml (hunk 3 above)
Comment (SandhyaRavi2403) [reply ansible/group_vars/all/defaults.yaml:18]: updated !!
https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.30.1

Comment (dkoshkin) [reply ansible/group_vars/all/defaults.yaml:18]: Thanks!

**Review by dkoshkin — APPROVED** [review]

**Pull request merged**